internal/auth/jwt.go

```diff
@@ -16,10 +29,42 @@ func NewTokenManager(publicKey, privateKey string) *TokenManager {
 
 // GenerateAccessToken generates new access token.
 func (tm *TokenManager) GenerateAccessToken(userID string) (string, error) {
-	return "", nil
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.StandardClaims{
+		Subject:   userID,
+		ExpiresAt: time.Now().Add(ACTimeout).Unix(),
+		IssuedAt:  time.Now().Unix(),
+	})
+
+	return token.SignedString(tm.privateKey)
 }
 
 // GenerateRefreshToken generates new refresh token.
 func (tm *TokenManager) GenerateRefreshToken() (string, error) {
-	return "", nil
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.StandardClaims{
+		ExpiresAt: time.Now().Add(RTTimeout).Unix(),
+		IssuedAt:  time.Now().Unix(),
+	})
+
+	return token.SignedString(tm.privateKey)
+}
+
+//ParseToken parses the given token and returns the user ID.
+func (tm *TokenManager) ParseToken(accessToken string) (string, error) {
+	token, err := jwt.Parse(accessToken, func(token *jwt.Token) (i interface{}, err error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+
+		return []byte(tm.privateKey), nil
+	})
+	if err != nil {
+		return "", err
+	}
+
+	claims, ok := token.Claims.(jwt.MapClaims)
```

Comment:
I guess, you can cast it to `jwt.StandardClaims` and return `claims.Subject` or even all claims. Thus, you won't need to case the subject specifically with `claims["sub"].(string)`

Reply:
> I guess, you can cast it to `jwt.StandardClaims` and return `claims.Subject` or even all claims. Thus, you won't need to case the subject specifically with `claims["sub"].(string)`

Thank you for advice!

Reply:
> I guess, you can cast it to `jwt.StandardClaims` and return `claims.Subject` or even all claims. Thus, you won't need to case the subject specifically with `claims["sub"].(string)`

Thank you for advice!